docs/source/external_exporters.rst

```diff
@@ -0,0 +1,156 @@
+.. _external_exporter
+
+Using external exporters
+========================
+
+.. versionadded:: 4.2
+
+    You can now use the ``--to`` flag to use custom export formats defined in other
+    libraries.
+
+
+The command-line syntax to run the ``nbconvert`` script is::
+
+  $ jupyter nbconvert --to FORMAT notebook.ipynb
+
+This will convert the Jupyter document file ``notebook.ipynb`` into the output
+format designated by the ``FORMAT`` string as explained below.
+
+A few built-in formats are available by default: `html`, `pdf`,
+`script`, `latex`. Each of these has its own _exporter_ with many configuration
+options that can be extended. Having the option to point to a different _exporter_ 
+allows authors to create their own fully customized templates. 
+
+A custom _exporter_ must be a globally importable python object. We recommend that
+these be distributed as python libraries.
+
+To export using an exporter from an external library, use the `fully qualified`
+name of this exporter on the command line as the argument of the ``--to`` flag::
```

Comment:
> Does it find/load regular setuptools entry points?

Yes.

> How does one declare the entrypoints?

Providers of entry points declare them in their packaging, using setuptools or flit (or other future packaging tools).

> Who needs to install it? Consumers of entrypoints? Providers of entrypoints? Both?

The entrypoints module is for consumers of entrypoints. If we used it like this, nbconvert would add a dependency on entrypoints.

> For us to rely on something like this in our core stack, I think it would need to be pretty stable and battle tested. How far aware is it from that?

I don't know if anyone has used it in earnest yet, but it's [documented](http://entrypoints.readthedocs.org/en/latest/) and [tested](https://travis-ci.org/takluyver/entrypoints). It's also only about 200 lines of code, so I'm pretty confident that we can quickly fix any problems we come across.